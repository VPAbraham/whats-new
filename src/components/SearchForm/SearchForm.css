header {
  width: 100%;
  border-bottom: white 1px solid;
  height: 10vh;
  display: flex;
  justify-content: space-around;
  align-items: center;
}

.header--h1 {
  color: white;
  font-size: 2.5em;
}

.header-span {
  color: #FEAE03;
}

.header--input {
  width: 60%;
  height: 50px;
  color: lightgray;
  background-color: #1f2150;
  border-radius: 10px;
  padding: 10px;
  border: 3px solid #25275c;
  font-size: 1.6em;
}

.header--button {
  background-color: #B56AF0;
  width: 230px;
  height: 50px;
  border-radius: 30px;
  border: none;
  color: white;
  font-size: 1em;
}