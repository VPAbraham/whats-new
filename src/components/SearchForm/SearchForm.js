import React, { Component } from 'react'
import './SearchForm.css'

class SearchForm extends Component{
  constructor() {
    super()
    this.state = {search: ''}
  }

  submitSearch = () => {
    this.props.filterArticles(this.state.search)
  }

  handleChange = (event) => {
    this.setState({ search: event.target.value })
    event.preventDefault()
  }

  render() {
    return (
      <header className="SearchForm">
        {console.log(this.props)}
        <h1 className="header--h1">What's 
          <span className="header--span"> New?</span>
        </h1>
        <input 
          className="header--input" 
          placeholder="Search for news article here."
          name="search"
          value={this.state.search}
          onChange={event => this.handleChange(event)}>
        </input>
        <button 
          className="header--button"
          onClick={event => this.submitSearch(event)}>
          Search Now
        </button>
      </header>
    )
  }
}
export default SearchForm