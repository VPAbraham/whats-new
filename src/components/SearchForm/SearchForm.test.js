import React from 'react'
import { shallow } from 'enzyme'
import SearchForm from './SearchForm'

describe('SearchForm', () => {
  let wrapper
  const mockFilterArticles = jest.fn()

  beforeEach(() => {
    wrapper = shallow(<SearchForm filterArticles = {mockFilterArticles}/>)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should be able to handle new text values', () => {
    const mockEvent = {target: { value: 'hello'}}
    wrapper.instance().handleChange = jest.fn()
    expect(wrapper.state('search')).toEqual('')
    wrapper.instance().handleChange(mockEvent)
    expect(wrapper.state('search')).toEqual('hello')

  })


})