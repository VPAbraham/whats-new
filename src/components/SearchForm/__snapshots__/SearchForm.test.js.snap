// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match the snapshot 1`] = `
<header
  className="SearchForm"
>
  <h1
    className="header--h1"
  >
    What's
    <span
      className="header--span"
    >
       New?
    </span>
  </h1>
  <input
    className="header--input"
    name="search"
    onChange={[Function]}
    placeholder="Search for news article here."
    value=""
  />
  <button
    className="header--button"
    onClick={[Function]}
  >
    Search Now
  </button>
</header>
`;
